### PR TITLE
Add keepEncodingHeader configuration

### PR DESCRIPTION
**Pull request by marcospereira**
## Purpose

Add `keepEncodingHeader` configuration, with default value of false.

## References

See discussion here: https://github.com/playframework/playframework/pull/7198#discussion_r110254631

### CODE REVIEW

File: play-ahc-ws-standalone/src/test/scala/play/api/libs/ws/ahc/AhcConfigBuilderSpec.scala [hunk 1]
```diff
@@ -156,6 +156,13 @@ class AhcConfigBuilderSpec extends Specification with Mockito {
         val actual = builder.build()
         actual.isDisableUrlEncodingForBoundRequests must_== true
       }
+
+      "allow setting ahc keepEncodingHeader" in {
+        val config = defaultConfig.copy(keepEncodingHeader = true)
+        val builder = new AhcConfigBuilder(config)
+        val actual = builder.build()
+        actual.isKeepAlive must_== true
```
Comment (wsargent) [inline play-ahc-ws-standalone/src/test/scala/play/api/libs/ws/ahc/AhcConfigBuilderSpec.scala]: Is this right?  Shouldn't this be keepEncodingHeader?

**Comment by domdorn** [comment]
if we keep the encoding-header, should we also have a way to disable the automatic decompression of the gzip content?

File: play-ahc-ws-standalone/src/test/scala/play/api/libs/ws/ahc/AhcConfigBuilderSpec.scala (hunk 1 above)
Comment (domdorn) [inline play-ahc-ws-standalone/src/test/scala/play/api/libs/ws/ahc/AhcConfigBuilderSpec.scala]: +1 .. isKeepAlive is something different

Comment (marcospereira) [inline play-ahc-ws-standalone/src/test/scala/play/api/libs/ws/ahc/AhcConfigBuilderSpec.scala]: Bad copy and paste. :-(

**Comment by marcospereira** [comment]
@wsargent and @domdorn please review it again.

I don't think we have a way to configure AHC to disable decompression. At least I couldn't find it.

**Comment by marcospereira** [comment]
Rebased against master.

**Comment by gmethvin** [comment]
So, as I understand it, AHC decodes the body automatically, so what is the purpose of looking at the `Content-Encoding` header? If I could get a raw body and decode it manually that'd be fine.

**Comment by marcospereira** [comment]
@gmethvin no use from the top of my head. The discussion started here: https://github.com/playframework/playframework/pull/7198#discussion_r110254631

But, since we are providing a AHC implementation for play-ws, I think it could be a good idea to give users the same configuration possibilities that they would have by using AHC directly.

**Pull request closed without merging**